ivadomed/loader/loader.py

```diff
@@ -197,12 +199,13 @@ def __init__(self, input_filenames, gt_filenames, metadata=None, slice_axis=2, c
         self.slice_axis = slice_axis
         self.soft_gt = soft_gt
         self.prepro_transforms = prepro_transforms
+        self.extension = input_filenames[0].split('.', 1)[1]
```

Comment:
Simply curious: why not using this info from the config file?

Reply:
> Simply curious: why not using this info from the config file?

Alternatively, we could just not store the extension at all. 

* `read_file` could parse the `filename` string.
* The other functions could check the type of the `data` variable. (`Nifti1Image` or `ndarray`)

Reply:
This is a very good question!
There are 3 places to fetch the extension:

1- From the config file:
As implemented right now, the extension param accepts multiple extensions as a list of string and not just single one. With multiple formats, we won't be able to tell which file has which extension. Also, the param is optional. If it is empty or not present, all files are indexed.

2 - From the metadata argument: 
That seems to be the best solution. It comes from the BIDSDataframe. However, it would not work for `segment_volume` in `inference.py`. This is use by `sct_deepseg` (if I'm not mistaken) and bypass the BIDS part of the loader entirely.

3 - From the filename:
That is what I implemented at the moment. I realize now, that it may not always work as intended with `segment_volume` if there are other dots in the filenames.

There may be a way to recognize the file type by its content without relying on the filename?

Reply:
@mariehbourget to follow-up on our meeting today, this is how we check file type in SCT, with additional checks based on the header (eg: we had bad surprises in the past, where files were NIfTI, but not images):

https://github.com/neuropoly/spinalcordtoolbox/blob/af2d4de531f66cd062af6fefa87c3f84185875ff/spinalcordtoolbox/scripts/sct_apply_transfo.py#L164-L169

Reply:
Fixed in 66b2e90